### ISSUE TITLE
re-running docker_extract.py errors when using singularity

### ISSUE THREAD

**Issue by johnbradley** [common-workflow-language/cwl-utils#12]
If you run `docker_extract.py --singularity ...` more than once with the same arguments it will fail with an error about `Image file already exists`. This halts execution and does not process any further docker images that are part of the workflow.

Example with stack trace(requires singularity to be installed):
```
$ docker_extract.py --singularity /tmp/singularity https://raw.githubusercontent.com/bespin-workflows/exomeseq-gatk4/develop/tools/fastqc.cwl

Traceback (most recent call last):
  File ".../cwl_utils/image_puller.py", line 28, in _run_command_pull
    stderr=subprocess.STDOUT)
  File ".../subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['singularity', 'pull', '--name', '/tmp/singularity/dukegcb_fastqc:0.11.4.sif', 'docker://dukegcb/fastqc:0.11.4']' returned non-zero exit status 255.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".../cwl_utils/docker_extract.py", line 72, in <module>
    sys.exit(main())
  File ".../cwl_utils/docker_extract.py", line 30, in main
    image_puller.save_docker_image()
  File ".../cwl_utils/image_puller.py", line 95, in save_docker_image
    ImagePuller._run_command_pull(cmd_pull)
  File ".../cwl_utils/image_puller.py", line 31, in _run_command_pull
    raise subprocess.SubprocessError(err.output)
subprocess.SubprocessError: b'\x1b[31mFATAL:  \x1b[0m Image file already exists: "/tmp/singularity/dukegcb_fastqc:0.11.4.sif" - will not overwrite\n'
```
The `Image file already exists` error seems to come from the `singularity` command. I think this is due to a difference between how the `docker` and `singularity` commands work. Without the `--singularity` flag the command runs fine multiple times without errors. I think cwltool has special logic to work around this difference between docker and singularity.

Thanks.

**Comment by mr-c** [comment]
Thanks @johnbradley 

Seems like docker is happy to overwrite an existing image and singularity does not want to.

What is your singularity version?

I would be happy to review a pull request to https://github.com/common-workflow-language/cwl-utils/blob/main/cwl_utils/image_puller.py#L90 that either skips existing images or replaces them. Probably should have a command line flag to choose the behavior with skipping + warning being the default.

**Comment by mr-c** [comment]
Fixed in #101
